.github/workflows/build-and-push-to-ghcr.yml

```diff
@@ -0,0 +1,48 @@
+name: Build and push to GHCR
+on:
+  push:
+    branches: [main]
+  pull_request:
+env:
+  GHCR_URL: ghcr.io/coqui-ai/xtts-streaming-server
+jobs:
+  build-and-push-to-ghcr:
+    runs-on: ubuntu-22.04
+    steps:
+      -
+        name: Checkout
+        uses: actions/checkout@v3
+
+      -
+        name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+
+      - name: 'Login to GitHub Container Registry'
+        run: |
+          set -xe
+          docker login --username ${{ github.actor }} --password ${{ secrets.GITHUB_TOKEN }} ghcr.io
+
+      - name: Build only for PR
+        if: github.ref != 'refs/heads/main'
+        uses: docker/build-push-action@v5
+        with:
+          context: "{{defaultContext}}:server"
+          file: Dockerfile
+          push: false # Do not push image for PR
+          cache-from: type=registry,ref=ghcr.io/coqui-ai/xtts-streaming-server:cache-pr-${{ github.event.pull_request.head.sha }}
```

Comment:
Also the from line should still pull from the main branch, to have some cache available on first push to a PR.